### PR TITLE
Bump dask in notebook image on publish so it doesn't fall behind the default software environment in coiled.

### PR DESCRIPTION
**Pull request by ian-r-rose**
Should fix coiled/cloud#1848

### CODE REVIEW

File: notebook/Dockerfile [hunk 1]
```diff
@@ -15,6 +15,7 @@ RUN conda create -y -n coiled \
     python=3.8 \
     "jupyterlab<3" \
     ipywidgets \
+    dask=2021.3.0 \
```
Comment (ian-r-rose) [inline notebook/Dockerfile:18]: This was implicitly included before, but making it explicit for the find/replace action

File: .github/workflows/ci-bump-dask.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        with:
+          org: "conda-forge"
+          package: "dask"
+
+      - name: Find and replace Dask version
+        id: find_and_replace
+        uses: jacobtomlinson/gha-find-replace@0.1.1
+        with:
+          find: "dask=.*"
+          replace: "dask=${{ steps.latest_version.outputs.version }} "
```
Comment (jrbourbeau) [inline .github/workflows/ci-bump-dask.yml:26]: I wonder if this will remove the trailing `\` character

File: .github/workflows/ci-bump-dask.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Output changed files
+        run: echo ${{ steps.find_and_replace.outputs.modifiedFiles }}
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          commit-message: "Update Dask version to ${{ steps.latest_version.outputs.version }}"
+          title: "Update Dask version to ${{ steps.latest_version.outputs.version }}"
+          reviewers: "jrbourbeau"
```
Comment (jrbourbeau) [inline .github/workflows/ci-bump-dask.yml:37]: No pressure

```suggestion
          reviewers: "jrbourbeau,ian-r-rose"
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @ian-r-rose!

File: .github/workflows/ci-bump-dask.yml (hunk 2 above)
Comment (ian-r-rose) [reply .github/workflows/ci-bump-dask.yml:26]: Good question, I can roll back the dask version bump here and we can inspect the resulting PR

Comment (jrbourbeau) [reply .github/workflows/ci-bump-dask.yml:26]: Sounds good 👍

**Review by jrbourbeau — APPROVED** [review]

**Pull request merged**